From c8416c66c2dd3cb7f16344e61236165578508d44 Mon Sep 17 00:00:00 2001
From: "Parncutt, Brandon" <Brandon.Parncutt@nielsen.com>
Date: Wed, 8 Dec 2021 13:05:45 -0500
Subject: [PATCH 1/5] tweaking showconfig output

---
 telegram_bot.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/telegram_bot.py b/telegram_bot.py
index c5614f4..44e7e72 100644
--- a/telegram_bot.py
+++ b/telegram_bot.py
@@ -433,7 +433,7 @@ class TelegramBot(TelegramBotBase):
         if query.data == "orders" or query.data == "pairs" or query.data == "allactive":
             self.marginresponse(update, context)
 
-        elif query.data in ("binance", "coinbasepro", "kucoin"):
+        elif query.data in ("binance", "coinbasepro", "kucoin", "scanner", "logging"):
             self.showconfigresponse(update, context)
 
         elif "pause_" in query.data:
@@ -866,7 +866,10 @@ class TelegramBot(TelegramBotBase):
 
         query = update.callback_query
 
-        pbot = self.config[query.data]["config"]
+        pbot = self.config[query.data]
+        pbot.delete('api_key')
+        pbot.delete('api_secret')
+        pbot.delete('api_passphrase')
 
         query.edit_message_text(query.data + "\n" + json.dumps(pbot, indent=4))
 
-- 
2.24.3 (Apple Git-128)


From 14606bb9109977233c12b82f06f6251132c6e298 Mon Sep 17 00:00:00 2001
From: "Parncutt, Brandon" <Brandon.Parncutt@nielsen.com>
Date: Thu, 9 Dec 2021 04:39:02 -0500
Subject: [PATCH 2/5] scanner upgrade

---
 scanner.json | 16 ++++++++++------
 scanner.py   | 41 ++++++++++++++++++++++++++++++++---------
 2 files changed, 42 insertions(+), 15 deletions(-)

diff --git a/scanner.json b/scanner.json
index bf30f50..1ad8dbe 100644
--- a/scanner.json
+++ b/scanner.json
@@ -4,17 +4,21 @@
 			"USD"
 		],
 		"tv_screener_ratings": [
-			"STRONG_BUY",
-			"BUY"
+			"STRONG_BUY"
 		],
 		"volatility_treshold": 9,
-		"volume_threshold": 20000,
+		"volume_threshold": 10000,
 		"adx_threshold": 25,
-		"granularity": 900
+		"granularity": 3600
 	},
 	"kucoin": {
 		"quote_currency": [
 			"USDT"
-		]
+		],
+		"volume_threshold": 10000,
+		"tv_screener_ratings": [
+			"STRONG_BUY"
+		],
+		"adx_threshold": 25
 	}
-}
\ No newline at end of file
+}
diff --git a/scanner.py b/scanner.py
index ff2f341..237af9a 100644
--- a/scanner.py
+++ b/scanner.py
@@ -118,7 +118,7 @@ def process_screener_data(app, markets, quote_currency):
     formatted_ta = []
     for ta in screener_analysis:
         try:
-            recommend = ta.summary.get('RECOMMENDATION')
+            recommend = Decimal(ta.indicators.get('Recommend.All'))
             volatility = Decimal(volatility_calculator(ta.indicators['BB.upper'], ta.indicators['BB.lower']))
             adx = abs(Decimal(ta.indicators['ADX']))
             adx_posi_di = Decimal(ta.indicators['ADX+DI'])
@@ -128,15 +128,22 @@ def process_screener_data(app, markets, quote_currency):
             macd_signal = Decimal(ta.indicators['MACD.signal'])
             bollinger_upper = Decimal(ta.indicators['BB.upper'])
             bollinger_lower = Decimal(ta.indicators['BB.lower'])
+            rsi = Decimal(ta.indicators.get('RSI', 0))
+            stoch_d = Decimal(ta.indicators.get('Stoch.D', 0))
+            stoch_k = Decimal(ta.indicators.get('Stoch.K', 0))
+            williams_r = Decimal(ta.indicators.get('W.R', 0))
             score = 0
             # print('symbol\tvolume\tvvolatilith\tadx\tadx_posi_di\tadx_neg_di\tmacd\tmacd_signal\tbollinger_upper\tbollinger_lower\trecommend')
             # print(ta.symbol, volume, volatility, adx, adx_posi_di, adx_neg_di, macd, macd_signal, bollinger_upper, bollinger_lower, recommend)
-            if recommend in app.tv_screener_ratings:
-                # print(ta.summary.get('RECOMMENDATION'))
+            if recommend <= 0.5 and recommend > 0.1:
+                score += 2.5
+                rating = 'BUY'
+            if recommend > 0.5:
                 score += 5
-            if (adx >= app.adx_threshold or adx_posi_di > adx) and (adx_posi_di > adx_neg_di):
+                rating = 'STRONG_BUY'
+            if ((adx >= app.adx_threshold) or (adx_posi_di > adx_neg_di)) and (adx_posi_di > adx):
                 # print(f"ADX({adx}) >= {app.adx_threshold}")
-                score += 2
+                score += 1 
             if volume >= app.volume_threshold:
                 # print(f"Volume({volume}) >= {app.volume_threshold}")
                 score += 1
@@ -146,6 +153,14 @@ def process_screener_data(app, markets, quote_currency):
             if volatility >= app.volatility_threshold:
                 # print(f"Volatility({volatility} is above {app.volatility_threshold}")
                 score += 1
+            if rsi <= 30 and rsi > 20:
+                score += 1
+            if stoch_d > 20 and stoch_d <= 30:
+                score += 1
+            if stoch_k >= stoch_d:
+                score += 1
+            if williams_r <= -30:
+                score += 1
             if score >= 10:
                 relavent_ta = {}
                 if app.exchange == CryptoExchange.COINBASEPRO or app.exchange == CryptoExchange.KUCOIN:
@@ -162,9 +177,13 @@ def process_screener_data(app, markets, quote_currency):
                 relavent_ta['macd.signal'] = macd_signal
                 relavent_ta['bollinger_upper'] = bollinger_upper
                 relavent_ta['bollinger_lower'] = bollinger_lower
-                relavent_ta['rating'] = recommend
+                relavent_ta['rsi'] = rsi
+                relavent_ta['stoch_d'] = stoch_d
+                relavent_ta['stoch_k'] = stoch_k
+                relavent_ta['williamsR'] = williams_r
+                relavent_ta['rating'] = rating
                 try:
-                    relavent_ta['buy_next'] = 'SEND IT!' if re.search('.*BUY', recommend).group() else False
+                    relavent_ta['buy_next'] = 'SEND IT!' if re.search('BUY', rating) else False
                 except AttributeError:
                     relavent_ta['buy_next'] = False
                 formatted_ta.append(relavent_ta)
@@ -173,8 +192,8 @@ def process_screener_data(app, markets, quote_currency):
     if formatted_ta:
         # Stick it in a DF for the bots
         df_markets = pd.DataFrame(formatted_ta)
-        df_markets = df_markets[["market", "volume", "volatility", "adx", "adx+di", "adx-di", "macd", "macd.signal", "bollinger_upper", "bollinger_lower", "rating", "buy_next"]]
-        df_markets.columns = ["market", "volume", "volatility", "adx", "adx+di", "adx-di", "macd", "macd.signal", "bollinger_upper", "bollinger_lower", "rating", "buy_next"]
+        df_markets = df_markets[["market", "volume", "volatility", "adx", "adx+di", "adx-di", "macd", "macd.signal", "bollinger_upper", "bollinger_lower", "rsi", "stoch_d", "stoch_k", "williamsR", "rating", "buy_next"]]
+        df_markets.columns = ["market", "volume", "volatility", "adx", "adx+di", "adx-di", "macd", "macd.signal", "bollinger_upper", "bollinger_lower", "rsi", "stoch_d", "stoch_k", "williamsR", "rating", "buy_next"]
         df_markets["volume"] = df_markets["volume"].astype(float).round(0).astype(int)
         df_markets["volatility"] = df_markets["volatility"].astype(float)
         df_markets["adx"] = df_markets["adx"].astype(float)
@@ -184,6 +203,10 @@ def process_screener_data(app, markets, quote_currency):
         df_markets["macd.signal"] = df_markets["macd.signal"].astype(float)
         df_markets["bollinger_upper"] = df_markets["bollinger_upper"].astype(float)
         df_markets["bollinger_lower"] = df_markets["bollinger_lower"].astype(float)
+        df_markets['rsi'] = df_markets['rsi'].astype(float)
+        df_markets['stoch_d'] = df_markets['stoch_d'].astype(float)
+        df_markets['stoch_k'] = df_markets['stoch_k'].astype(float)
+        df_markets['williamsR'] = df_markets['williamsR'].astype(float)
         df_markets.sort_values(by=["market"], ascending=True, inplace=True)
         df_markets.set_index("market", inplace=True)
 
-- 
2.24.3 (Apple Git-128)


From 0f5f9b331f6b445e7a21f3f9e2479d25a7d549aa Mon Sep 17 00:00:00 2001
From: "Parncutt, Brandon" <Brandon.Parncutt@nielsen.com>
Date: Fri, 10 Dec 2021 19:01:07 -0500
Subject: [PATCH 3/5] criteria tweaks; fixed missing key error on config
 sanitization

---
 .gitignore      | 3 ++-
 scanner.py      | 4 ++--
 telegram_bot.py | 9 ++++++---
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/.gitignore b/.gitignore
index 15a50bd..fcf430f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -152,4 +152,5 @@ tests/unit_tests/data/*.json
 
 *.key
 config-*.yaml
-telegram_data*
\ No newline at end of file
+telegram_data*
+scanner_*.zip
diff --git a/scanner.py b/scanner.py
index 237af9a..b7bf795 100644
--- a/scanner.py
+++ b/scanner.py
@@ -141,7 +141,7 @@ def process_screener_data(app, markets, quote_currency):
             if recommend > 0.5:
                 score += 5
                 rating = 'STRONG_BUY'
-            if ((adx >= app.adx_threshold) or (adx_posi_di > adx_neg_di)) and (adx_posi_di > adx):
+            if (adx >= app.adx_threshold) and (adx_posi_di > adx_neg_di) and (adx_posi_di > adx):
                 # print(f"ADX({adx}) >= {app.adx_threshold}")
                 score += 1 
             if volume >= app.volume_threshold:
@@ -157,7 +157,7 @@ def process_screener_data(app, markets, quote_currency):
                 score += 1
             if stoch_d > 20 and stoch_d <= 30:
                 score += 1
-            if stoch_k >= stoch_d:
+            if stoch_k > stoch_d:
                 score += 1
             if williams_r <= -30:
                 score += 1
diff --git a/telegram_bot.py b/telegram_bot.py
index 44e7e72..1e8b548 100644
--- a/telegram_bot.py
+++ b/telegram_bot.py
@@ -867,9 +867,12 @@ class TelegramBot(TelegramBotBase):
         query = update.callback_query
 
         pbot = self.config[query.data]
-        pbot.delete('api_key')
-        pbot.delete('api_secret')
-        pbot.delete('api_passphrase')
+        try:
+            pbot.delete('api_key')
+            pbot.delete('api_secret')
+            pbot.delete('api_passphrase')
+        except Exception as e:
+            pass
 
         query.edit_message_text(query.data + "\n" + json.dumps(pbot, indent=4))
 
-- 
2.24.3 (Apple Git-128)


From 5e53e712e843705349220e653fbb7b5546dd99c5 Mon Sep 17 00:00:00 2001
From: "Parncutt, Brandon" <Brandon.Parncutt@nielsen.com>
Date: Sun, 12 Dec 2021 04:40:18 -0500
Subject: [PATCH 4/5] bug fix in dual comparisons

---
 scanner.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/scanner.py b/scanner.py
index b7bf795..dbc852f 100644
--- a/scanner.py
+++ b/scanner.py
@@ -135,7 +135,7 @@ def process_screener_data(app, markets, quote_currency):
             score = 0
             # print('symbol\tvolume\tvvolatilith\tadx\tadx_posi_di\tadx_neg_di\tmacd\tmacd_signal\tbollinger_upper\tbollinger_lower\trecommend')
             # print(ta.symbol, volume, volatility, adx, adx_posi_di, adx_neg_di, macd, macd_signal, bollinger_upper, bollinger_lower, recommend)
-            if recommend <= 0.5 and recommend > 0.1:
+            if 0.5 >= recommend > 0.1:
                 score += 2.5
                 rating = 'BUY'
             if recommend > 0.5:
@@ -153,9 +153,9 @@ def process_screener_data(app, markets, quote_currency):
             if volatility >= app.volatility_threshold:
                 # print(f"Volatility({volatility} is above {app.volatility_threshold}")
                 score += 1
-            if rsi <= 30 and rsi > 20:
+            if 30 >= rsi > 20:
                 score += 1
-            if stoch_d > 20 and stoch_d <= 30:
+            if 20 < stoch_d <= 30:
                 score += 1
             if stoch_k > stoch_d:
                 score += 1
@@ -188,7 +188,7 @@ def process_screener_data(app, markets, quote_currency):
                     relavent_ta['buy_next'] = False
                 formatted_ta.append(relavent_ta)
         except Exception as e:
-            continue
+            pass
     if formatted_ta:
         # Stick it in a DF for the bots
         df_markets = pd.DataFrame(formatted_ta)
-- 
2.24.3 (Apple Git-128)


From 9d2ad170a1e0475a0bbb896beb67117a8aec69d5 Mon Sep 17 00:00:00 2001
From: "Parncutt, Brandon" <Brandon.Parncutt@nielsen.com>
Date: Sun, 12 Dec 2021 10:52:51 -0500
Subject: [PATCH 5/5] fixing bug that creeped in from beta

---
 models/config/binance_parser.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/models/config/binance_parser.py b/models/config/binance_parser.py
index ff838ef..adf6334 100644
--- a/models/config/binance_parser.py
+++ b/models/config/binance_parser.py
@@ -9,7 +9,7 @@ from .default_parser import isCurrencyValid, defaultConfigParse, merge_config_an
 def isMarketValid(market) -> bool:
     if market == None:
         return False
-    p = re.compile(r"^[0-9A-Z]{1,10}\-[1-9A-Z]{2,5}$")
+    p = re.compile(r"^[0-9A-Z]{1,10}[1-9A-Z]{2,5}$")
     if p.match(market):
         return True
     return False
-- 
2.24.3 (Apple Git-128)

